addons/test_mail/tests/test_mail_composer.py

```diff
@@ -453,6 +453,37 @@ def test_mail_composer_rights_attachments(self):
             sorted(self.test_record.message_ids[0].attachment_ids.mapped('name')),
             sorted(template_1_attachment_name))
 
+    @users('employee')
+    def test_mail_composer_template_switching(self):
+        """ Ensure that the current user's identity serves as the sender,
+        particularly when transitioning from a template with a designated sender to one lacking such specifications.
+        Moreover, we verify that switching to a template lacking any subject maintains the existing subject intact.
+        """
+        # Setup: Prepare Templates
+        template_with_sender_subject = self.template.copy({
```

Comment:
Maybe, if someday we add some more variations (more fields to check), we could use a more generic naming like
```suggestion
        template_complete = self.template.copy({
```